graphqlapi/src/main/java/io/stargate/graphql/web/resources/schemafirst/FilesResource.java

```diff
@@ -38,6 +50,27 @@ public Response getCqlDirectives() {
         .build();
   }
 
+  @GET
+  @Path("/namespace/{namespaceName}.graphql")
+  @Produces(MediaType.TEXT_PLAIN)
+  public Response getSchema(
+      @PathParam("namespaceName") String namespace,
+      @QueryParam("version") String version,
+      @Context HttpServletRequest httpRequest)
+      throws Exception {
+    SchemaSource schemaSource =
+        schemaSourceDao.getByVersion(namespace, Optional.ofNullable(version).map(UUID::fromString));
+
+    return Response.ok(schemaSource.getContents())
+        .header("Content-Disposition", "inline; filename=" + createFileName(schemaSource))
+        .build();
+  }
+
+  private String createFileName(SchemaSource schemaSource) {
+    return String.format(
+        "\"%s-%s.graphql\"", schemaSource.getNamespace(), schemaSource.getVersion());
```

Comment:
I think that it may be UX friendly if we will include both namespace and version in the schema name. Wdyt?

Reply:
Yes, that's what I had in mind 👍